### ISSUE TITLE
Eclim Always Reports No Errors

### ISSUE THREAD

**Issue by vanderheijden86** [emacs-eclim/emacs-eclim#78]
## Expected behavior
Eclim will report error at text "asdfasfd"
## Actual behavior
![image](https://user-images.githubusercontent.com/16118904/33274974-dcab0f92-d391-11e7-8f0c-91ed05227419.png)

## Steps to reproduce the problem

- Install Emacs 25.3.1
- Install Eclipse Oxygen 3.1.a
- Install Eclim
- Run Emacs and start-eclimd

## Environment & Version information
Mac OS X Sierra 10.12.6
Emacs 25.3.1
Eclipse Oxygen 3.1.a
Eclim 2.7.0

### package version information
Eclim             20171113.1754 

### Java Development Kit Version

JDK 1.8

### Eclipse version

Oxygen

### Eclim version

Eg. 2.7.0

### Emacs version

25.3.1

### Operating system

OS X Sierra 10.12.6

**Comment by nloyola** [comment]
I cannot reproduce this on my system. Can you please try the following? 

1. Modify your configuration to have `(setq eclim-print-debug-messages t)` and exit Emacs.
1. Start the Eclim server (`eclimd`) in a terminal shell (outside of Emacs)
1. Start Emacs
1. Open the `.java` file with the syntax error

Paste the logging from in the terminal window. Paste the logging in the `*Messages*` buffer.

**Comment by vanderheijden86** [comment]
Thanks for digging into this with me. 

***Messages* buffer:**
`Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command ping
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command project_by_resource -f /Users/avdh/Documents/codex1_chain_code/src/main/java/tech/b3i/codex1/Codex1.java [3 times]
eclimd serving at port 9091
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command project_list
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command project_info -p codex1_chain_code
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command project_link_resource -f /Users/avdh/Documents/codex1_chain_code/src/main/java/tech/b3i/codex1/Codex1.java
Saving file /Users/avdh/Documents/codex1_chain_code/src/main/java/tech/b3i/codex1/Codex1.java...
Wrote /Users/avdh/Documents/codex1_chain_code/src/main/java/tech/b3i/codex1/Codex1.java
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command project_by_resource -f /Users/avdh/Documents/codex1_chain_code/src/main/java/tech/b3i/codex1/Codex1.java
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command java_src_update -p codex1_chain_code -f src/main/java/tech/b3i/codex1/Codex1.java
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command java_complete -p codex1_chain_code -f src/main/java/tech/b3i/codex1/Codex1.java -e iso-8859-1 -l standard -o 164
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command problems -p codex1_chain_code
Using async buffer *eclim-async*
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command workspace_dir
Executing: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/bin/eclim -command project_by_resource -f /Users/avdh/Documents/codex1_chain_code/src/main/java/tech/b3i/codex1/Codex1.java
Eclim reports 0 errors, 0 warnings.
Auto-saving...done
`

**Terminal window** (Stays empty)
![image](https://user-images.githubusercontent.com/16118904/33447477-b5be11f8-d603-11e7-8793-2d89ac7daa87.png)


**eclimd** buffer in Emacs
`/usr/bin/java -version
java version "1.8.0_151"
Java(TM) SE Runtime Environment (build 1.8.0_151-b12)
Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)
JAVA_VERSION=8

/usr/bin/java -d64 -Dosgi.instance.area.default=@user.home/eclipse-workspace/ -Xmx256M -XstartOnFirstThread -jar /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.equinox.launcher_1.4.0.v20161219-1356.jar --launcher.suppressErrors -debug -clean -refresh -application org.eclim.application
Install location:
    file:/Applications/Eclipse.app/Contents/Eclipse/
Configuration file:
    file:/Applications/Eclipse.app/Contents/Eclipse/configuration/config.ini loaded
Configuration location:
    file:/Applications/Eclipse.app/Contents/Eclipse/configuration/
Framework located:
    file:/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.osgi_3.12.50.v20170928-1321.jar
Loading extension: reference:file:org.eclipse.osgi.compatibility.state_1.1.0.v20170516-1513.jar
	eclipse.properties not found
Framework classpath:
    file:/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.osgi_3.12.50.v20170928-1321.jar
    file:/Applications/Eclipse.app/Contents/Eclipse/plugins/
    file:/Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclipse.osgi.compatibility.state_1.1.0.v20170516-1513.jar
Debug options:
    file:/Users/avdh/Documents/codex1_chain_code/src/main/java/tech/b3i/codex1/.options not found
Time to load bundles: 43
org.eclipse.m2e.logback.configuration: The org.eclipse.m2e.logback.configuration bundle was activated before the state location was initialized.  Will retry after the state location is initialized.
Starting application: 2520
org.eclipse.m2e.logback.configuration: Logback config file: /Users/avdh/eclipse-workspace/.metadata/.plugins/org.eclipse.m2e.logback.configuration/logback.1.8.2.20171007-0217.xml
SLF4J: Class path contains multiple SLF4J bindings.
SLF4J: Found binding in [bundleresource://502.fwk1476394199:1/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: Found binding in [bundleresource://502.fwk1476394199:2/org/slf4j/impl/StaticLoggerBinder.class]
SLF4J: See http://www.slf4j.org/codes.html#multiple_bindings for an explanation.
SLF4J: Actual binding is of type [ch.qos.logback.classic.util.ContextSelectorStaticBinder]
org.eclipse.m2e.logback.configuration: Initializing logback
Application Started: 5166
2017-11-30 19:18:06,746 INFO  [org.eclim.eclipse.EclimDaemon] Workspace: /Users/avdh/eclipse-workspace
2017-11-30 19:18:06,748 INFO  [org.eclim.eclipse.EclimDaemon] Home: /Applications/Eclipse.app/Contents/Eclipse/plugins/org.eclim_2.7.0/
2017-11-30 19:18:06,748 INFO  [org.eclim.eclipse.EclimDaemon] Starting eclim...
2017-11-30 19:18:06,757 INFO  [org.eclim.eclipse.EclimDaemon] Loading plugin org.eclim
2017-11-30 19:18:06,760 INFO  [org.eclim.eclipse.EclimDaemon] Loading plugin org.eclim.core
2017-11-30 19:18:07,158 INFO  [org.eclim.plugin.core.CorePlugin] Loading eclim plugins...
2017-11-30 19:18:07,162 INFO  [org.eclim.plugin.core.CorePlugin] Loading plugin org.eclim.jdt
2017-11-30 19:18:07,163 INFO  [org.eclim.plugin.core.CorePlugin] Plugins loaded.
2017-11-30 19:18:07,164 INFO  [org.eclim.eclipse.EclimDaemon] Loaded plugin org.eclim.core
2017-11-30 19:18:07,165 INFO  [org.eclim.eclipse.EclimDaemon] Waiting on running jobs before starting eclimd...
2017-11-30 19:18:12,265 INFO  [org.eclim.eclipse.EclimDaemon] Jobs finished.
2017-11-30 19:18:12,266 INFO  [org.eclim.eclipse.EclimDaemon] Eclim Server Started on: 127.0.0.1:9091
`

**Comment by nloyola** [comment]
I'd like to ensure that the problem is present in **Emacs Eclim**. One way to figure this out is to try the following outside of Emacs.

You can run the Eclim server and client from the command line. For me these two executables were installed into the same directory where the Eclipse executable is:

```
/home/nelson/apps/eclipse/java/oxygen/eclipse/eclimd
/home/nelson/apps/eclipse/java/oxygen/eclipse/eclim
```
I created a small test project, called **javatest**, in `/home/nelson/src/test/javatest` where I put a syntax error in the Java file `/home/nelson/src/test/javatest/src/javatest/Test.java`.

I can start the Eclim server with this command:

```
<eclipse_path>/eclimd -Dosgi.instance.area.default=/home/nelson/src/test
```


*Where `<eclipse_path>` is the directory Eclipse was installed in.*

Once I see the following log message, I know that the Eclim server is ready:

```
2017-11-30 12:54:55,979 INFO  [org.eclim.eclipse.EclimDaemon] Eclim Server Started on: 127.0.0.1:9091
```

Then I can run the Eclim client to list the errors in my source using this command:

```
<eclipse_path>/eclim -command problems -p javatest
```
The client generates the following output:

```
[{"message":"Syntax error, insert \"Identifier (\" to complete MethodHeaderName","filename":"/home/nelson/src/test/javatest/src/javatest/Test.java","line":5,"column":5,"endLine":-1,"endColumn":-1,"warning":false},{"message":"Syntax error, insert \")\" to complete MethodDeclaration","filename":"/home/nelson/src/test/javatest/src/javatest/Test.java","line":5,"column":5,"endLine":-1,"endColumn":-1,"warning":false},{"message":"Syntax error, insert \";\" to complete ClassBodyDeclarations","filename":"/home/nelson/src/test/javatest/src/javatest/Test.java","line":5,"column":5,"endLine":-1,"endColumn":-1,"warning":false}]
```
Can you please try this for your project?

**Comment by nloyola** [comment]
Sorry, I pressed the wrong button and closed this by accident.

**Comment by vanderheijden86** [comment]
All right, did just that. For me this works as well:
![image](https://user-images.githubusercontent.com/16118904/33457996-c6db187a-d624-11e7-8e0b-d8e8a705ad0f.png)

For this project, it **does work** in Emacs as one would expect. The other project I experienced the absence of error checking with was not in my eclipse-workspace, I cloned it in another folder (as I never work with Eclipse). So this is the reason for the error then I guess? Sorry you had to go and dig into this for me to discover this to be the reason, just didn't figure this had to be the case...
